bigquery_etl/cli/query.py

```diff
@@ -2199,12 +2203,22 @@ def deploy(
         query_files = paths_matching_name_pattern(
             name, ctx.obj["TMP_DIR"], project_id, ["query.*", "script.sql"]
         )
-        if not query_files:
+        metadata_files = paths_matching_name_pattern(
+            name, ctx.obj["TMP_DIR"], project_id, ["metadata.yaml"]
+        )
+        if not query_files and not metadata_files:
             raise click.ClickException(f"No queries matching `{name}` were found.")
 
     credentials = get_credentials()
     id_token = get_id_token(credentials=credentials)
 
+    query_file_paths = [query_file.parent for query_file in query_files]
+    metadata_files_without_query_file = [
+        metadata_file
+        for metadata_file in metadata_files
+        if metadata_file.parent not in query_file_paths
+    ]
```

Comment:
Wouldn't this also pick up the metadata files of `view.sql` and then deploy them as tables if they have schemas?

Reply:
mh yeah, I guess for newly added views that don't already exist that would happen...

Reply:
Added a filter to filter out metadata files for which views and materialized views exist